mlrun/datastore/sources.py

```diff
@@ -866,8 +866,8 @@ def __init__(
         key_field: str = None,
         time_field: str = None,
         schedule: str = None,
-        start_time=None,
-        end_time=None,
+        start_time: Optional[Union[datetime, str]] = None,
```

Comment:
the type hint Optional is mostly for `dataclass` objects so the methods will be set accordingly. When you have `= None` it is optional by default :)